pkg/controller/servicebindingrequest/secret_test.go

```diff
@@ -29,18 +39,7 @@ func TestSecretNew(t *testing.T) {
 	)
 
 	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data)
-		assert.NoError(t, err)
-		assertSecretNamespacedName(t, u, ns, name)
-	})
-
-	t.Run("Delete", func(t *testing.T) {
-		err := s.delete()
-		assert.NoError(t, err)
-	})
```

Comment:
we still have `secrete.delete()` in the prod code, but not used anywhere else. Please remote it then as well.

Reply:
This PR is just about adding owner reference to secret.
Please create an issue for other changes related to secret.

Reply:
> This PR is just about adding owner reference to secret.

The PR is about implementing secret removal using k8s garbage collection mechanism, and adding owner reference is an implementation detail. With that approach,, we introduced some dead code in projec, i.e. `secret.delete()`. Housekeeping should be done within this PR.

Reply:
Got it. Done here c7be281